src/diff.rs

```diff
@@ -74,51 +75,89 @@ pub struct DiffStats {
     raw: *mut raw::git_diff_stats,
 }
 
+/// Structure describing the binary contents of a diff.
+pub struct DiffBinary<'diff> {
+    raw: *const raw::git_diff_binary,
+    _marker: marker::PhantomData<&'diff Diff<'diff>>,
+}
+
+/// The contents of one of the files in a binary diff.
+pub struct DiffBinaryFile<'diff> {
+    raw: *const raw::git_diff_binary_file,
+    _marker: marker::PhantomData<&'diff Diff<'diff>>,
```

Comment:
Is this ok or do I need to do something like

```rust
pub struct DiffBinary<'diff, 'repo: 'diff> {
  _marker: PhantomData<&'diff Diff<'repo>>,
}
```

I tried a couple of changes to the test that I thought might break with this and they didn't so I assume there's some sort of greater than relationship automatically happening with the lifetimes here?